### PR TITLE
Update README.md's to use "contribution bar"

### PR DESCRIPTION
**Pull request by steveharter**
An update to https://github.com/dotnet/runtime/pull/76567 based on various discussions.

Instead of using "Active", "Inactive" and "Legacy" for the "Status", change wording to "contribution bar" that more accurately describes what type of PRs and issues are expected for each library. The contribution bar essentially is the acceptable risk and investment level for each library given the current status of the library and is composed of a single primary bar and optional secondary bars.

cc @joperezr per our offline review

### CODE REVIEW

File: src/libraries/README.md [hunk 1]
```diff
@@ -2,22 +2,37 @@
...
 
-- **Active**
-  - Under active development by the team
-  - Issues will be considered for fix or addition to the backlog
+### Assumed bars (unless a library says otherwise)
+- **We take security fixes**
+- **We take test changes**
+- **We don't take style-only changes**
+
+### Primary bar
```
Comment (steveharter) [inline src/libraries/README.md:14]: If a library has a different composition, the thinking is it doesn't necessarily need to update this list and doesn't necessarily need to link back to here.

File: src/libraries/README.md [hunk 2]
```diff
@@ -2,22 +2,37 @@
...
+- **We take new features, new APIs and performance changes**
+  - This is the most encompassing category and takes on the most risk
```
Comment (joperezr) [inline src/libraries/README.md:16]: Looks like there is a trailing space on this line which will fail CI

**Review by jeffhandley — APPROVED** [review]
This is great. Thank you for pioneering this for us.

My only _suggestion_ is to change "we take" to "we consider" and "we don't take" to "we don't accept."

File: src/libraries/README.md [hunk 3]
```diff
@@ -2,22 +2,37 @@
...
+- **We only take lower-risk or high-impact fixes to maintain or improve quality**
+  - New features and APIs are **not** considered
+  - We don't take potentially destabilizing fixes or test changes unless there is a clear need
+- **We only take fixes that unblock critical issues**
+  - New features and APIs are **not** considered
+  - This may include infrastructure changes and other housekeeping
+
+### Secondary bars
+- **We take PRs that target this library for new [source code analyzers](../../docs/project/analyzers.md)**
+- **We don't take refactorting changes due to new language features**
```
Comment (danmoseley) [inline src/libraries/README.md:35]:
```suggestion
- **We don't take refactoring changes due to new language features**
```

File: src/libraries/README.md [hunk 4]
```diff
@@ -2,22 +2,37 @@
...
+Some libraries are under more active development than others. Depending on the library's status, expectations for issues and pull requests can vary. Check the library's folder for a `README.md` that declares the contribution bar for that library which consists of  a **Primary bar** and optional **Secondary bars**. Regardless of a library's contribution bar, refer to the [DOs and DON'Ts](../../CONTRIBUTING.md#dos-and-donts) and [Suggested Workflow](../../CONTRIBUTING.md#suggested-workflow) in our contribution guidelines before submitting a pull request.
 
-- **Active**
-  - Under active development by the team
-  - Issues will be considered for fix or addition to the backlog
+### Assumed bars (unless a library says otherwise)
+- **We take security fixes**
+- **We take test changes**
```
[the remark this answers is not in the document]
Comment (steveharter) [reply src/libraries/README.md]: Done. Thanks

File: src/libraries/System.Reflection.Metadata/README.md [hunk 5]
```diff
@@ -1,8 +1,11 @@
 # System.Reflection.Metadata
 Documentation can be found at https://learn.microsoft.com/dotnet/api/system.reflection.metadata.
 
-## Status: [Inactive](../../libraries/README.md#development-statuses)
-The APIs and functionality are mature, but do get extended occasionally.
+## Contribution Bar
+- [x] [We take new features, new APIs and performance changes](../../libraries/README.md#primary-bar)
```
[the remark this answers is not in the document]
Comment (steveharter) [reply src/libraries/System.Reflection.Metadata/README.md]: There was such a bullet for that, but it was slightly tweaked in the latest commit:
`The library is likely mature and feature-complete but may be extended occasionally`

File: src/libraries/README.md (hunk 3 above)
Comment (steveharter) [reply src/libraries/README.md]: Yikes thanks

**Review by joperezr — APPROVED** [review]

**Pull request merged**